src/builder/QuarkBuilder.sol

```diff
@@ -574,8 +573,24 @@ contract QuarkBuilder {
         if (payment.isToken) {
             uint256 supplementalPaymentTokenBalance = 0;
             if (Strings.stringEqIgnoreCase(payment.currency, cometWithdrawIntent.assetSymbol)) {
-                // XXX in the withdrawMax case, use the Comet balance
-                supplementalPaymentTokenBalance += cometWithdrawIntent.amount;
+                if (isMaxWithdraw) {
+                    // when doing a maxWithdraw of the payment token, add the account's supplied balance
+                    // as supplemental payment token balance
+                    //
+                    // assumes that a maxWithdraw is for the base asset, since
+                    // you cannot maxWithdraw Collateral
+                    Accounts.CometPosition memory cometPosition = Accounts.findCometPosition(
+                        cometWithdrawIntent.chainId, cometWithdrawIntent.comet, chainAccountsList
+                    );
+
+                    for (uint256 i = 0; i < cometPosition.basePosition.accounts.length; ++i) {
+                        if (cometPosition.basePosition.accounts[i] == cometWithdrawIntent.withdrawer) {
+                            supplementalPaymentTokenBalance += cometPosition.basePosition.supplied[i];
```

Comment:
here we're using `basePosition.supplied` to get the best guess of how much of the asset the user will end up with ultimately.

i'm assuming that `basePosition.supplied` is best way piece of information we have about the current withdrawable amount, but I don't know if that number reflects how much of the base asset has been supplied or the amount supplied + interest accrued

```
"cometPositions" => [
    %{
      "comet" => "0x2943ac1216979ad8db76d9147f64e61adc126e96",
      "borrowAPR" => 9945224338320000,
      "supplyAPR" => 4793472000,
      "borrowRewardsAPR" => 0,
      "supplyRewardsAPR" => 1699636062375893083,
      "base" => %{
        "asset" => "0x2d5ee574e710219a521449679a4a7f2b43f046ad",
        "decimals" => 18,
        "usdPrice" => 315288151686,
        "totalBorrow" => 969017000499,
        "totalSupply" => 3405644257260925148,
        "totalBalance" => 3405643280745890843,
        "accounts" => [
          "0xf6F148cFb92dD3baF399B0eCD1a8561b7f16487D",
          "0xE276Bc378A527A8792B353cdCA5b5E53263DfB9e"
        ],
        "borrowed" => [0, 0],
        "supplied" => [0, 0],
        "usdBorrowCapacity" => [0, 0],
        "baseBorrowCapacity" => [0, 0],
        "usdLiquidationPoint" => [0, 0],
        "baseLiquidationPoint" => [0, 0]
      },
      "collaterals" => [
        %{
          "asset" => "0xb9fa8f5ec3da13b508f462243ad0555b46e028df",
          "decimals" => 18,
          "usdPrice" => 305829507135,
          "basePrice" => 97000000,
          "totalSupply" => 2000134115105271783,
          "accounts" => [
            "0xf6F148cFb92dD3baF399B0eCD1a8561b7f16487D",
            "0xE276Bc378A527A8792B353cdCA5b5E53263DfB9e"
          ],
          "balances" => [0, 0]
        },
      ]
    }
 ]
```

Reply:
I assume it includes interest accrued, as there's no way to get the original supplied amount in the current block.